Diff/diff.py

```diff
@@ -45,120 +47,162 @@ def run(self, files):
 
     def is_enabled(self):
         return self.view.is_dirty() and self.view.file_name() is not None
 
 
-def show_diff_output(diff, view, win, name, panel_name, buffer_setting_name):
+def show_diff_output(
+    diff,
+    view: typing.Union[None, sublime.View],
+    window: sublime.Window,
+    name: str,
+    panel_name: str,
+    buffer_setting_name: str
+):
     difftxt = u"".join(line for line in diff)
 
     if difftxt == "":
         sublime.status_message("No changes")
         return
 
-    use_buffer = not view or view.settings().get(buffer_setting_name)
+    use_buffer: bool = not view or view.settings().get(buffer_setting_name)
 
     if use_buffer:
-        v = win.new_file()
-        v.set_name(name)
-        v.set_scratch(True)
+        new_view: sublime.View = window.new_file()
+        new_view.set_name(name)
+        new_view.set_scratch(True)
     else:
-        v = win.create_output_panel(panel_name)
+        new_view = window.create_output_panel(panel_name)
         if view:
-            v.settings().set('word_wrap', view.settings().get('word_wrap'))
+            new_view.settings().set(
+                'word_wrap',
+                view.settings().get('word_wrap')
+            )
 
-    v.assign_syntax('Packages/Diff/Diff.sublime-syntax')
-    v.run_command('append', {'characters': difftxt, 'disable_tab_translation': True})
+    new_view.assign_syntax('Packages/Diff/Diff.sublime-syntax')
+    new_view.run_command(
+        'append',
+        {'characters': difftxt, 'disable_tab_translation': True}
+    )
 
     if not use_buffer:
-        win.run_command('show_panel', {'panel': f'output.{panel_name}'})
+        window.run_command('show_panel', {'panel': f'output.{panel_name}'})
 
 
-def get_view_from_tab_context(active_view, **kwargs):
-    view = active_view
+def get_view_from_tab_context(
+    active_view: sublime.View,
+    **kwargs
+):
+    view: sublime.View = active_view
     if 'group' in kwargs and 'index' in kwargs:
-        view = view.window().views_in_group(kwargs['group'])[kwargs['index']]
+        window: typing.Optional[sublime.Window] = view.window()
+        if window is not None:
+            view = window.views_in_group(kwargs['group'])[kwargs['index']]
     return view
 
 
-def get_views_from_tab_context(active_view, **kwargs):
-    selected_views = list(get_selected_views(active_view.window()))
-    if 'group' in kwargs and 'index' in kwargs:
-        tab_context_view = get_view_from_tab_context(active_view, **kwargs)
-        # if the tab which was right clicked on is selected, exclude it from the selected views and re-add it afterwards
-        # so that the order of the diff will be determined by which tab was right-clicked on
-        return [view for view in selected_views if view.id() != tab_context_view.id()] + [tab_context_view]
+def get_views_from_tab_context(
+    active_view: sublime.View,
+    **kwargs
+) -> typing.Union[None, typing.List[sublime.View]]:
```

Comment:
It would simplify code to always return a list but empty, if needed. Unless there is specific need to differentiate those cases.